### DISCUSSION TITLE
Feature Request: allow pinning more than just 6 repos to home page

### DISCUSSION THREAD

**Discussion by HariSekhon** [community/community#28350] [Other Feature Feedback, Questions, & Ideas]
I have a lot of very highly relevant repositories and would like to be able to pin more than 6 repos to my home page.

6 seems like an arbitrarily small number.

Why not increase the limit to at least 8 or 10 or 12?

It's not like it would even change the structure of a user's profile page, it would just be one or two more rows of repos listed as my core ones that I want somebody landing on my profile page to see.

**Comment by jfullstackdev** [comment]
you can use an external service, and add as many as you want 

https://github.com/anuraghazra/github-readme-stats#github-extra-pins

**Reply by timmyjose** [reply]
Disappointing to see that this is still not addressed.

**Comment by HariSekhon** [comment]
So I finally got round to working around this using a personal repo README to replace the top portion of my github profile page:

https://github.com/HariSekhon

You can do anything you normally can in markdown, format the page, use HTML and pull in 3rd party tools to embed stats or reproduce those GitHub repo blocks to list a whole bunch of them without this arbitrary 6 repo pin limitation.

I've added a documentation on this and some interested projects you may want to use to create your GitHub homepage here:

https://github.com/HariSekhon/Knowledge-Base/blob/main/github.md#github-profile-page

**Reply by timmyjose** [reply]
That actually looks quite nice!

**Comment by ufukty** [comment]


**Reply by vycdev** [reply]
> Limiting the number of pins beyond how many repository one user can create is unnecessary. It should be unlimited. As the pins section is the only place we can list our repositories with **persistent order**, it makes us save time while navigating between repositories.

I also find it very unnecessary. I have to use [github-readme-stats](https://github.com/anuraghazra/github-readme-stats) to add extra repository "pins" in my GitHub README. Which looks pretty bad because of the limitations of the README and the inconsistencies in styling.

**Comment by taublast** [comment]
For a nice flex grid with unlimited repos inside your README:

(replace `username` and `repo0`, `repo1`..)

```xml
<div style="display:flex; flex-wrap:wrap; gap:12px; align-items:flex-start;">
  <a href="https://github.com/username/repo0" style="flex:1 1 320px; max-width:420px;">
    <img src="https://github-readme-stats-fast.vercel.app/api/pin/?username=username&repo=repo0&bg_color=0d1117&title_color=58a6ff&text_color=c9d1d9&icon_color=58a6ff&border_color=30363d&hide_border=false&border_radius=10&description_lines_count=2" />
  </a>

  <a href="https://github.com/username/repo1" style="flex:1 1 320px; max-width:420px;">
    <img src="https://github-readme-stats-fast.vercel.app/api/pin/?username=username&repo=repo1&bg_color=0d1117&title_color=58a6ff&text_color=c9d1d9&icon_color=c9d1d9&border_color=30363d&hide_border=false&border_radius=10&description_lines_count=2" />
  </a>

  <!-- repeat for the rest -->
</div>
```
